app/javascript/authentication/SignInForm.tsx

```diff
@@ -98,4 +128,48 @@ const SignInForm = () => {
   )
 }
 
+const onSubmit =
+  (setNotConfirmed: (email: string) => void, setRequestError: (error: string) => void) =>
+  (data: { email: string; password: string }) => {
+    const { email, password } = data
+
+    signIn(email, password)
+      .then(() => {
+        window.location.href = getMyAccountPath()
+        window.scrollTo(0, 0)
+      })
+      .catch((error: AxiosError<{ error: string; email: string }>) => {
+        if (error.response.data.error === "not_confirmed") {
+          setNotConfirmed(error.response.data.email)
+        } else {
+          // TODO: handle sign-in error states
+          setRequestError(`${t("signIn.badCredentials")}`)
+        }
+      })
+  }
+
+const SignInForm = () => {
+  const [requestError, setRequestError] = useState<string>()
+  const [notConfirmed, setNotConfirmed] = useState<string | null>(null)
```

Comment:
Suggestion: Make this naming more specific, perhaps `showNotConfirmedModal`. Just so it's clear if someone is looking at in the future what this state value is used for